lib/providers/generic.js

```diff
@@ -40,51 +40,50 @@ class GenericProvider {
   }
 
   /**
-   * Retrieve the generic secret.
-   *
-   * The callback has the following signature: `callback(err, data)`.
-   *
-   * @param {function} callback
+   * Retrieve the secret.
+   * @returns {Promise}
    */
-  initialize(callback) {
-    if (this._value) {
-      callback(new Error('Already initialized'), null);
-      return;
-    }
-
-    this._client.prepare(this.token).then(() => this._retrieve(callback));
-  }
+  initialize() {
+    return new Promise((resolve, reject) => {
+      if (this._value) {
+        resolve(this.renew());
+        return;
+      }
 
-  _retrieve(callback) {
-    return this._client[this._method]({id: this.path, token: this.token})
-      .then(this._read.bind(this, callback))
-      .catch((err) => callback(err, null));
+      this._retrieve()
+        .then((data) => resolve(data))
+        .catch((err) => reject(err));
+    });
   }
 
   /**
-   * Renew the generic secret.
-   *
-   * The method name is effectively the same as initializing the secret.
-   *
-   * The callback has the following signature: `callback(err, data)`.
-   *
-   * @param {function} callback
+   * Retrieve the requested secret from Vault
+   * @returns {Promise}
+   * @private
    */
-  renew(callback) {
-    this._retrieve(callback);
+  _retrieve() {
+    return new Promise((resolve, reject) => {
+      this._client.prepare(this.token)
+        .then(() => {
+          this._client[this._method]({id: this.path, token: this.token})
```

Comment:
Needs a `return` inside of a `{}`, otherwise the wrapping Promise won't chain to it.